lib/api.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const setupContext = require('./context');
+const Widget = require('./widget');
+const _ = require('lodash');
+
+module.exports = {createClient};
+
+function createClient (options) {
+  options = options || {};
+  const context = setupContext({
+    token: options.accessToken,
+    host: options.host
+  });
+
+  return {get: get, getAll, save, delete: del};
+
+  function get (id) {
```

Comment:
I'm my opinion having the functions defined after the `return` line make the code less readable as when I get to the `return` statement I still have no knowledge about the properties that are being included in the returned object: they could be values, functions or some unassigned variable. But to know that I have to keep reading and then go back to the `return` statement to check that everything looks correct.